test/Microsoft.Health.Events.UnitTest/EventConsumerServiceTests.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.Health.Events.EventConsumers;
+using Microsoft.Health.Events.EventConsumers.Service;
+using Microsoft.Health.Events.Model;
+using Microsoft.Health.Logging.Telemetry;
+using NSubstitute;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Xunit;
+
+namespace Microsoft.Health.Events.UnitTest
+{
+    public class EventConsumerServiceTests
+    {
+        [Fact]
+        public async void GivenEventConsumer_WhenEventConsumerThrowsException_ThenRetryEventConsumer_Test()
+        {
+            var retries = 0;
+            var initialBatch = new List<EventMessage>()
+            {
+                new EventMessage("0", new ReadOnlyMemory<byte>(), 1, 1, new DateTime(2020, 12, 31, 5, 10, 20), new Dictionary<string, object>(), new ReadOnlyDictionary<string, object>(new Dictionary<string, object>()))
```

Comment:
Nit, not sure if the EventMessage supports parameterless constructor so we can use the object initializer with sending properties to make this a little more readable.